api/v1/stats/service.go

```diff
@@ -0,0 +1,210 @@
+	aggAppendSum(s, c, "server stats subscription sends")
+}
+
+func (s *Service) GetAppendInRecords(c *gin.Context) {
+	aggAppendSum(s, c, "server stats stream append_in_record")
+}
+
+func (s *Service) GetSendOutRecords(c *gin.Context) {
+	aggAppendSum(s, c, "server stats subscription send_out_records")
+}
+
+func (s *Service) GetAppendTotal(c *gin.Context) {
+	aggAppendSum(s, c, "server stats stream_counter append_total")
+}
+
+func (s *Service) GetAppendFailed(c *gin.Context) {
+	aggAppendSum(s, c, "server stats stream_counter append_failed")
+}
+
+func (s *Service) GetServerAppendRequestLatency(c *gin.Context) {
+	aggSum(s, c, "server stats server_histogram append_request_latency")
+}
+
+func (s *Service) GetServerAppendLatency(c *gin.Context) {
+	aggSum(s, c, "server stats server_histogram append_latency")
+}
+
+func adminError(err error) errorno.ErrorResponse {
+	return errorno.NewErrorResponse(errorno.ADMIN_GET_STATUS_ERROR, err)
+}
+
+func getFromCluster(s *Service, cmd string) ([]model.TableResult, error) {
+	info, err := s.client.GetServerInfo()
+	if err != nil {
+		return nil, fmt.Errorf("GetServerInfo error: %v", err)
+	}
+
+	allStatsRaw := make([]model.TableResult, 0)
+
+	for _, x := range info {
+		err := func() error {
+			client, err := hstream.NewHStreamClient(x)
+			if err != nil {
+				return fmt.Errorf("NewHStreamClient error with %v: %v", x, err)
+			}
+			defer client.Close()
+
+			resp, err := client.AdminRequest(cmd)
+			if err != nil {
+				return fmt.Errorf("AdminRequest error with %v: %v", cmd, err)
+			}
+
+			var jsonObj map[string]json.RawMessage
+			if err := json.Unmarshal([]byte(resp), &jsonObj); err != nil {
+				return fmt.Errorf("unmarshal to RawMessage error: %v", err)
+			}
+
+			var table model.TableType
+			if content, ok := jsonObj["content"]; ok {
+				if err = json.Unmarshal(content, &table); err != nil {
+					return fmt.Errorf("unmarshal to TableType error: %v", err)
+				}
+			} else {
+				return fmt.Errorf("JSON object does not have the `content` field")
+			}
+
+			stats := model.TableResult{
+				Headers: table.Headers,
+				Value:   make([]map[string]string, 0, len(table.Rows)),
+			}
+
+			for _, row := range table.Rows {
+				mp := make(map[string]string, len(row))
+				for idx, val := range row {
+					mp[table.Headers[idx]] = val
+				}
+				stats.Value = append(stats.Value, mp)
+			}
+
+			allStatsRaw = append(allStatsRaw, stats)
+			return nil
+		}()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return allStatsRaw, nil
+}
+
+func aggAppendSum(s *Service, c *gin.Context, cmd string) {
+	allStatsRaw, err := getFromCluster(s, cmd)
+	if err != nil {
+		c.AbortWithStatusJSON(http.StatusInternalServerError, adminError(err))
+		return
+	}
+
+	if len(allStatsRaw) != 0 {
+		headers := allStatsRaw[0].Headers
+
+		allStats := model.TableResult{
+			Headers: headers,
+			Value:   make([]map[string]string, 0),
+		}
+
+		for _, stats := range allStatsRaw {
```

Comment:
Can you explain the logic of aggregation with some real examples? It seems that this implementation may contain some bugs.